bundles/org.eclipse.e4.ui.workbench.renderers.swt/src/org/eclipse/e4/ui/workbench/renderers/swt/CTabRendering.java

```diff
@@ -91,7 +88,7 @@ public class CTabRendering extends CTabFolderRenderer implements ICTabRendering,
 	public static final boolean SHOW_FULL_TEXT_FOR_VIEW_TABS_DEFAULT = false;
 
 	private static int MIN_VIEW_CHARS = 1;
-	private static int MAX_VIEW_CHARS = 9999;
+	private static int MAX_VIEW_CHARS = 999999;
```

Comment:
Please either stop using a magic number or explain and document in the code why  exactly the huge value 999999, rather than 999998, 1000000, or Integer.MAX_VALUE.  Especially because MAX_VALUE makes it clear that there is effectively no minimum.

Reply:
I agree. I have set _Integer.MAX_VALUE_ as MAX_VIEW_CHARS instead of any other large number. This constant is itself self explanatory and safe guards in case a tab with really large title text is possible (Mega screen).